File: src/serverless/BlazingPizza.Functions.FSharp/OrderById.fs
namespace BlazingPizza.Functions.FSharp

open Microsoft.AspNetCore.Mvc
open Microsoft.Azure.WebJobs
open Microsoft.Azure.WebJobs.Extensions.Http
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.Logging

open BlazingPizza
open System.Security.Claims

module OrderById =
    [<FunctionName("order-by-id")>]
    let run ([<HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = "orders/{orderId:int}")>] req: HttpRequest)
            ([<CosmosDB("blazingPizza",
                        "pizza",
                        ConnectionStringSetting = "CosmosConnectionString",
                        SqlQuery = "SELECT * FROM c WHERE c.OrderId = {orderId}")>] requestedOrder: Order seq)
            (orderId: int)
            (log: ILogger)
            =
        let principal = ClientPrincipal.parse req

        let userId =
            principal.Claims
            |> Seq.find (fun c -> c.Type = ClaimTypes.NameIdentifier)

        let order = requestedOrder |> Seq.tryHead

        match order with
        | Some order when order.UserId = userId.Value ->
            OkObjectResult(OrderWithStatus.FromOrder(order)) :> IActionResult
        | Some _
        | None -> NotFoundResult() :> IActionResult

File: src/serverless/BlazingPizza.Functions.CSharp/OrderById.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using System.Collections.Generic;
using System.Linq;
using System.Security.Claims;

namespace BlazingPizza.Functions.CSharp
{
    public static class OrderById
    {
        [FunctionName("order-by-id")]
        public static IActionResult Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "orders/{orderId:int}")] HttpRequest req,
            [CosmosDB("blazingPizza",
                      "pizza",
                      ConnectionStringSetting = "CosmosConnectionString",
                      SqlQuery = "SELECT * FROM c WHERE c.OrderId = {orderId}")] IEnumerable<Order> requestedOrder,
            int orderId,
            ILogger log)
        {
            var principal = StaticWebAppsAuth.Parse(req);
            var userId = principal.Claims.First(c => c.Type == ClaimTypes.NameIdentifier);

            var order = requestedOrder.FirstOrDefault(order => order.UserId == userId.Value);

            if (order == null)
            {
                return new NotFoundResult();
            }

            return new OkObjectResult(OrderWithStatus.FromOrder(order));
        }
    }
}
